### ISSUE TITLE
Inconsistent formatting when log records contain data

### ISSUE THREAD

**Issue by eikek** [outr/scribe#295]
Hi, I'm not sure if I misunderstood… in my log output all lines end with `()` which looked strange. I'm not aware that I customized this (using log format "enhanced") so i looked around and found this:

There is code building a map out of two and checking the result for being empty or not. After being checked for emptiness it is not used to evaluate, but only one of the maps:

https://github.com/outr/scribe/blob/141f0e0a35eebd0ba96a846d94b76ed59b500122/core/shared/src/main/scala/scribe/format/FormatBlock.scala#L267-L269

This also applies to other location in this file. I think this could be the cause why I'm seeing the `()` at the end of my log lines? My MDC is empty, but I have data in the log record sometimes.

**Comment by darkfrog26** [comment]
Can you show some code that reproduces the problem?

**Comment by eikek** [comment]
Sure, sorry. I might very well have misconfigured something. I'll try to create a minimum example.

**Comment by hindog** [comment]
I just hit this issue while integration scribe into another framework (ZIO).  In my case, I adapt each ZIO log to a Scribe `LogRecord` and pass ZIO's fiber-local `LogContext` map as the `data` parameter, but the values aren't coming through.  Here's an example to demonstrate the issue:

```
import scribe._
import scribe.data.MDC

object LogExample extends App{

  MDC.update("key1", "value1")

  log(LogRecord(
    Level.Info,
    Level.Info.value,
    new LazyMessage(() => "Some message"),
    implicitly[Loggable[String]],
    None,
    "file.scala",
    "Class",
    None,
    None,
    None,
    data = Map(
      "key2" -> (() => "value2")
    )
  ))
}

# prints:
2022.05.19 18:04:39:660 [main      ] [INFO ] Class - Some message (key1: value1)

# expected:
2022.05.19 18:04:39:660 [main      ] [INFO ] Class - Some message (key1: value1, key2: value2)
```

I believe the fix is to change the code @eikek linked to something like this:
```
 val map = MDC.map ++ record.data 
 if (map.nonEmpty) { 
   new TextOutput(map.map { 
```

**Comment by eikek** [comment]
Sorry I forgot to respond - thank you for the example @hindog 🙏🏼

**Comment by darkfrog26** [comment]
Thanks for the code example. I'll take a look at this and get back to you ASAP.

**Comment by darkfrog26** [comment]
Okay, this is fixed and I'm doing a release now for `3.8.3` that will include it.

**Comment by eikek** [comment]
I can confirm it fixes my issue, thanks!
